src/main/java/thaumicenergistics/common/container/ContainerDistillationPatternEncoder.java

```diff
@@ -362,11 +343,14 @@ protected boolean detectAndSendChangesMP(@Nonnull final EntityPlayerMP playerMP)
             return true;
         }
         // Does the source item need to be sync'd?
-        else if (this.slotSourceItem.getStack() != this.cachedSource) {
-            // Scan the source item
-            this.scanSourceItem(true);
-            return true;
-        }
+        else if (this.cachedSource == null || this.slotSourceItem.getStack() == null
+                || (this.cachedSource != null && this.slotSourceItem.getStack() != null
+                        && !this.slotSourceItem.getStack().getDisplayName()
```

Comment:
It's better and faster to compare items and metadata instead of comparing display names.
Also, two first conditions in parentheses are redundant.